### PR TITLE
DDCE-410 new-added messages for contact details

### PR DESCRIPTION
**Pull request by prameya-phatak**
# DDCE-(410-new)

## New feature 

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/forms/contact/CharityContactDetailsFormProvider.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import utils.CharitiesValidator
+import play.api.data.Form
+import play.api.data.Forms._
+
+class CharityContactDetailsFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[CharityContactDetails] =
+    Form(
+      mapping(
+        "mainPhoneNumber" -> text("charityContactDetails.mainPhoneNumber.error.format")
```
Comment (anburoach) [inline app/forms/contact/CharityContactDetailsFormProvider.scala:31]: error messages on line 31 and 32 should be replaced

File: app/utils/CharitiesValidator.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+object CharitiesValidator {
```
Comment (anburoach) [inline app/utils/CharitiesValidator.scala:19]: As we discussed... please remove this file and move the 2 variables to CharityContactDetailsFormProvider.scala

File: conf/messages.en [hunk 3]
```diff
@@ -153,13 +153,28 @@ inEligible.p3.link = check the HMRC charities and tax guidance (opens in new win
...
+
+
+# CharityContactDetailsPage Messages
+# ----------------------------------------------------------
+charityContactDetails.title = Charity''s contact details
+charityContactDetails.heading = Charity''s contact details
+charityContactDetails.checkYourAnswersLabel = Charity''s contact details
+charityContactDetails.mainPhoneNumber.label = Main phone number
+charityContactDetails.mainPhoneNumber.error.format = Enter your charity''s daytime phone number, like 01632 960 001, 07700 900 982 or +44 0808 157 0192
+charityContactDetails.mainPhoneNumber.error.required = Enter your charity''s daytime telephone number, like 01632 960 001, 07700 900 982 or +44 0808 157 0192
```
Comment (anburoach) [inline conf/messages.en:175]: Error message is mismatch with story.... plz update the message

File: test/forms/contact/CharityContactDetailsFormProviderSpec.scala [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+class CharityContactDetailsFormProviderSpec extends StringFieldBehaviours {
+
+  val maxLength = 160
+
+  val form = new CharityContactDetailsFormProvider()()
+
+  ".mainPhoneNumber" must {
+
+    val requiredKey = "charityContactDetails.mainPhoneNumber.error.format"
```
Comment (anburoach) [inline test/forms/contact/CharityContactDetailsFormProviderSpec.scala:31]: Change the message key to charityContactDetails.mainPhoneNumber.error.required

File: test/navigation/ContactNavigatorSpec.scala [hunk 5]
```diff
@@ -40,7 +40,7 @@ class ContactNavigatorSpec extends SpecBase{
         "go to the InEligibleController page when No is selected" in {
```
Comment (anburoach) [inline test/navigation/ContactNavigatorSpec.scala:40]: modify this to valid test message

File: app/forms/contact/CharityContactDetailsFormProvider.scala (hunk 1 above)
Comment (prameya-phatak) [reply app/forms/contact/CharityContactDetailsFormProvider.scala]: modified

File: app/utils/CharitiesValidator.scala (hunk 2 above)
Comment (prameya-phatak) [reply app/utils/CharitiesValidator.scala]: removed this file and moved variables to form

**Pull request merged**